### ISSUE TITLE
Cant access a lot of UI elements from addons

### ISSUE THREAD

**Issue by qualitykk** [sboxgame/issues#2893]
### Describe the bug

In an addon, if you try to use most built-in UI elements it gives an error about the class not existing even when importing everything correctly. This only seems to happen in c# and seems to work just fine in razor however.

UI Elements which have been tested and work
- Panel
- Label

UI Elements which have been tested and dont work
- ChatBox
- Button
- IconPanel

### To Reproduce

1. Create/Open an Addon
2. Create a UI element in C# code
3. Add required using
4. Save/Build the addon

### Expected behavior

The addon should compile as expected and allow you to use those UI elements.

### Media/Files

![image](https://user-images.githubusercontent.com/57487634/217864589-1da03f73-88e0-4879-aacc-67a293e89046.png)
![image](https://user-images.githubusercontent.com/57487634/217864690-87a8cd6f-3d5e-48cf-a9f1-88c3475b8683.png)

**Comment by qualitykk** [comment]
Yep after checking all the UI types i checked that didnt work were in base which would explain why some work and some dont, I guess fixing it shouldnt be too hard then.

**Comment by lotuspar** [comment]
to add on to this - if you patch the compiler to reference base yourself your addon can go onto asset.party fine - not sure why this isn't already a thing

https://asset.party/gio/baseaddontest

**Comment by ShadowBrian** [comment]
This seems fairly essential for making elaborate addons with their own UI elements, pretty odd it's not a thing already.

**Comment by DevulTj** [comment]
I pushed something that should hopefully solve this. I'm a little on the fence about it, but we'll now drag in Base when publishing addons, as well as use your local Base when developing addons locally.

**Comment by chrisspieler** [comment]
As of now, this is not working. In an addon project that uses the WorldPanel type from the Base Library, I'm getting errors about a missing using directive or assembly, even though I have `using Sandbox.UI;` at the top of my file. Others have reported similar issues with using Base Library code.

<img width="513" alt="image" src="https://user-images.githubusercontent.com/8889285/230356802-7c513fe8-073f-48d1-95ac-581ce08670e5.png">

**Comment by DevulTj** [comment]
Is this still the case?

**Comment by chrisspieler** [comment]
> Is this still the case?

Yes
